### PR TITLE
Add flexibility for installables

### PR DESCRIPTION
**Pull request by bish0polis**
## Description

In a world where the same binaries are provided by different installable artifacts, this tiny patch allows someone to specify the name of said binary delivery wrapper.

### Check List
- [ ] All tests pass. See https://github.com/sous-chefs/mariadb/blob/master/TESTING.md
- [ ] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable

### CODE REVIEW

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This Pull Request is probably missing tests.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: resources/client_install.rb [hunk 1]
```diff
@@ -18,8 +18,12 @@
 
 property :version,    String, default: '10.3'
 property :setup_repo, [true, false], default: true
+property :package_name, String, default: 'MariaDB-client'
 
 action :install do
+  node.run_state['mariadb'] ||= {}
+  node.run_state['mariadb']['client_pkg'] = new_resource.package_name
```
Comment (xorima) [inline resources/client_install.rb:25]: Hey,

Why are we storing this inside the `run_state`?

**Review by xorima — CHANGES_REQUESTED** [review]

File: resources/server_install.rb [hunk 2]
```diff
@@ -27,17 +27,20 @@
 property :password,          [String, nil], default: 'generate'
 property :port,              Integer,       default: 3306
 property :initdb_locale,     String,        default: 'UTF-8'
+property :package_name,      String,        default: node['platform_family'] == 'debian' ? "mariadb-server-#{new_resource.version}" : 'MariaDB-server'
```
Comment (xorima) [inline resources/server_install.rb]: Inside the property definition there is no variable called `new_resource` only the raw property names as variables, so this should be: 
```suggestion
property :package_name,      String,        default: node['platform_family'] == 'debian' ? "mariadb-server-#{version}" : 'MariaDB-server'
```

**Review by xorima — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,16 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](http://keepachangelog.com/) and this project adheres to [Semantic Versioning](http://semver.org/).
 
+## [3.1.1] (2019-11-16)
```
Comment (xorima) [inline CHANGELOG.md:7]:
```suggestion
## Unreleased
```

**Review by xorima — CHANGES_REQUESTED** [review]

**Comment by xorima** [comment]
Hey @bish0polis 

Can you just check the tests related to this change and make sure they work locally, I know we have a few issues here with tests so want to ensure things are working

If you need a hand drop by the #sous-chefs channel in https://community-slack.chef.io/

Thanks

**Comment by xorima** [comment]
Hey @bish0polis 

Do you have time to work on this? It would be great to have in the code 

Thanks

Sous Chefs

**Comment by damacus** [comment]
## Closing due to inactivity. 

If this is still an issue please reopen or open another PR. 

Alternatively drop by the #sous-chefs channel on the [Chef Community Slack](http://community-slack.chef.io/) and we'll be happy to help! 

Thanks,
Sous-Chefs

**Comment by bish0polis** [comment]
Sorry for the delay. I'll reopen this later, but I'll only be testing on cinc because of license issues.

**Pull request closed without merging**